### PR TITLE
New blerg post

### PR DESCRIPTION
**Pull request by bcardarella**
I feel I might be all over the place with this one. Please let me know

### CODE REVIEW

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+comments: true
+author: 'Brian Cardarella'
+twitter: 'bcardarella'
+social: true
+published: true
+tags: elixir
+---
+
+I have been programming for over 20 years now. I started with Basic,
+found my way to C++, sepnt two years writing Assembly (MASM). Then I
```
Comment (duggiefresh) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: *spent*

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+found Ruby. Ruby completely changed everything for me. I loved Ruby. I
+loved Ruby for a reason that many "elite" programmers tend to dismiss:
+the syntax.
+
+You see, syntax is very important to me. Call it what you will,
+bikeshedding, OCD, stupidity. I care about syntax. It matters to me, and
+with Ruby I found a community that shared my thoughts.
```
Comment (duggiefresh) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: "and with Ruby*,* I found..."

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+When Go and Rust came along I was disappointed. Clearly these two
+languages were superior in performance (and in many other areas) but were a syntatic step back
```
Comment (duggiefresh) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: *syntactic*

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md (hunk 1 above)
Comment (rsocci) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: **and** spent

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+talk about parallel processing the state of a process becomes very
+important. If we are expecting `X` to always be a specific value but we
+are writing in a language where `X` can change this can lead to
+problems.
+
+## 4. Fault Tolerance
+
+This one really impressed me when I started to dig into it. You may have
+heard that Erlang was invented for telephony. How often do you get a
+message from your phone company saying "we're updating our systems so
```
Comment (michaeldupuisjr) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: Love the "we're updating our systems" analogy. Conveys fault tolerance well.

**Comment by rsocci** [comment]
I think it’s great. I like the different points you make about why one should look into using it, and it doesn’t feel like too much or too little info about each one

**Comment by michaeldupuisjr** [comment]
@bcardarella I don't think the post rambled at all. You state you were attracted to Elixir's syntax, disappointed in the other functional programming alternatives, and then explain what else you learned to love about Elixir.

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md (hunk 4 above)
Comment (bcardarella) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: I wish I could take credit for that one, I've heard it several times so I stole it

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+own memory, if the process is short-lived enough GC is never even run
+and the process is destroyed after it has completed its job. As opposed
+to Ruby where everything lives in one world and if you stop using the
+object it will get GC'd eventually impacting performance.
+
+## 3. Immutability
+
+Immutability got a bad rap when memory was expensive. Why would we write
+applications in such a way so as to waste memory by having variables
+who's values couldn't be mutated. Memory is now super cheap, and this is
```
Comment (kidfribble) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: should be a question mark "...couldn't be mutated?"

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+You see, syntax is very important to me. Call it what you will,
+bikeshedding, OCD, stupidity. I care about syntax. It matters to me, and
+with Ruby I found a community that shared my thoughts.
+
+When Go and Rust came along I was disappointed. Clearly these two
+languages were superior in performance (and in many other areas) but were a syntatic step back
+from Ruby. What was their reason? Clearly Ruby, and even Python, have
+proven that the masses are attracted by clear and readable syntax. New
+languages should take the best of what is currently availabale and
```
Comment (poteto) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: *available*

File: source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+there is a new blog article on why you should start writing Functional
+code. Let's break this down into a few points:
+
+## 1. Scalability
+
+This is an Erlang trait. Elixir apps will attempt to make the best use
+of all the cores in your CPU as possible. Compared to Ruby this is a big
+deal. We don't have to write anything special, the Erlang VM (BEAM) just
+handles this for us automatically. This means we are effeciently using
+our hardware. This type of approached didn't make a lot of sense a few
```
Comment (poteto) [inline source/posts/2015-04-01-elixir-come-for-the-syntax-stay-for-everything-else.md]: *approach*

**Pull request merged**